.github/runs-on.yml

```diff
@@ -0,0 +1,8 @@
+runners:
+  self-hosted-ubuntu22-full-x64:
```

Comment:
Still, why calling it `self-hosted-ubuntu22-full-x64` and not more canonical `self-hosted-ubuntu-22.04-x86-64`?

This is completely inconsistent with anything we had before, so I'm wondering why. Like is there non-`full` version? Why no separator between `ubuntu` and `22`, etc? Let's make it consistent an actually make sense.

Also there was a different image before, what was it and why do we have a different one now?

Reply:
I kept it aligned with the image name from the provider. I think "makes sense" is subjective here but I'll change it.